Bottom: 9a6c97c626b351d8b2486959957b9a128781f4cf
Top:    9047916ac482a52eb998715177bd853f57733611
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-14 02:33:05 +0000

Refresh of tag-circular-contours-so-they-

---

diff --git a/src/hid/common/draw_helpers.c b/src/hid/common/draw_helpers.c
index 5c49d3e..0da8c0a 100644
--- a/src/hid/common/draw_helpers.c
+++ b/src/hid/common/draw_helpers.c
@@ -29,13 +29,20 @@ static void thindraw_contour (hidGC gc, PLINE *pl)
   int last_x, last_y;
   int this_x, this_y;
 
+  gui->set_line_width (gc, 0);
+  gui->set_line_cap (gc, Round_Cap);
+
+  /* If the contour is round, use an arc drawing routine. */
+  if (pl->is_round)
+    {
+      gui->draw_arc (gc, pl->cx, pl->cy, pl->radius, pl->radius, 0, 360);
+      return 0;
+    }
+
   /* Need at least two points in the contour */
   if (pl->head.next == NULL)
     return;
 
-  gui->set_line_width (gc, 0);
-  gui->set_line_cap (gc, Round_Cap);
-
   last_x = pl->head.point[0];
   last_y = pl->head.point[1];
   v = pl->head.next;
@@ -177,13 +184,6 @@ static int thindraw_hole_cb (PLINE *pl, void *user_data)
 {
   hidGC gc = user_data;
 
-  /* If the contour is round, use an arc drawing routine. */
-  if (pl->is_round) {
-    gui->set_line_width (gc, 0);
-    gui->draw_arc (gc, pl->cx, pl->cy, pl->radius, pl->radius, 0, 360);
-    return 0;
-  }
-
   thindraw_contour (gc, pl);
   return 0;
 }
